cloud/aws/aws_file_system.h

```diff
@@ -19,6 +19,12 @@
 #include <unordered_map>
 
 namespace ROCKSDB_NAMESPACE {
+
+// Initializes AWS, or joins an existing initialization. Keep the
+// returned shared_ptr alive for as long as you will be using AWS
+// operations
+std::shared_ptr<void> useAWS();
```

Comment:
How common is the pattern of using shared_ptr to indicate global resource lifecycle? Would the API be nicer as a an RAII pattern? Don't know if it's feasible in this case though.

Reply:
Agree in principal, but usage differs. I assume this code will be invoked like:

```
// DONT REMOVE - variable protects AWS client resources
auto unused_variable = useAWS()
```

As opposed to RAII class that would read as

```
aws_client_guard guard()
```

I'm not used to seeing the `unused_variable` approach much, but the gaurd case is so common. Though, I don't don't do much C++ :P.

Reply:
Every single `lock_guard` over a mutex is an unused variable :) It's a very common C++ idiom. It would be better if an instance of the guard variable was necessary to do aws operations, but the amount of plumbing we would need to enforce that would be insane.